cooltools/lib/common.py

```diff
@@ -506,3 +508,39 @@ def align_track_with_cooler(
         clr_track.loc[~valid_bins, "value"] = np.nan
 
     return clr_track[["chrom", "start", "end", "value"]]
+
+
+def pool_decorator(func):
+    """
+    A decorator function that enables multiprocessing for a given function.
+
+    Parameters
+    ----------
+    func : callable
+        The function to be decorated.
+
+    Returns
+    -------
+    A wrapper function that enables multiprocessing for the given function.
+    """
+
+    def wrapper(*args, **kwargs):
+        pool = None
+        if "nproc" in kwargs.keys():
+            if kwargs["nproc"] > 1:
+                pool = Pool(kwargs["nproc"])
+                mymap = pool.map
```

Comment:
`mp.Pool.map` is not always what you want to use in the decorated function. It is **eager** and will [materialize all outputs in a list](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.pool.Pool.map).

This is problematic if the function you are mapping is not reductive (e.g. when making a pixel chunk iterator, all chunks will be materialized in memory). Instead you want to use a lazier map implementation like `imap` or in some cases `imap_unordered`.

Reply:
option for a test for this function could be returning what map is being used by it...

Reply:
seems like cooler indeed used pool.imap for tabix loader: https://github.com/open2c/cooler/blob/857601826fb3aa8ae4e4d3cc64afa61dec18c87e/src/cooler/cli/cload.py#L222

Reply:
pool.map was used for balance b/c the length bintable is reasonable to hold in memory, as are various function outputs from cooltools